### DISCUSSION TITLE
Some questions about docking (version xtb-6.6.0)

### DISCUSSION THREAD

**Discussion by codecollider** [grimme-lab/xtb#779] [Q&A]
Dear all,

I´m currently using the docking algorithm of xtb to get structures of cyclodextrin and POM adducts (1 POM and 1 to several cyclodextrins) as CREST takes too much time for these systems. To generate higher aggregates I first dock the first cyclodextrine to the POM and the structure looks reasonably good and is more stable than any "manually" generated structure. The problem comes in when I try to add the second cyclodextrin. 

Shown on the left is what I expect and shown on the right is the outcome from xtb (I know that the left structure is more stable than the right). The POM is schematically drawn in blue and the cyclodextrin in orange. 

--> Link to imgur with schematic drawing:

i.imgur.com/XiUVvRl.jpg

I assume that due to the limited number of starting orientations  of the POM*CD unit and the CD unit the expected geometry is not found. Does this make any sense? Is there any parameter I can adjust to have more and possibly better resolved  starting orientations? I tried also directed docking by defining prefered O atoms on the POM. It is still running, but from what I see it does not help. 

Another thing that concerns me is that I get a lot of NaN energies in the genetic optimization algorithm (see below as an example output). 

The command I used  to execute the docking is: 

xtb dock mol1_chrg_min6.xyz mol2_chrg_neutral.xyz --gfn 1 --maxgen 20

mol1_chrg_min6.xyz = POM*CD
mol2_chrg_neutral.xyz = CD

Charges I defined in .CHRG. 

I also wonder were the temporary files are stored when doing a docking run. I have a opt_tmp file which has 0 bytes and one final_opt.xyz file that only includes the most recent calculation. 

Thanks for any help!
Marco
---------------------------------------------
`Precomputation of electronic porperties
  For Molecule 1
  Successful
  For Molecule 2
  Successful
  LUMO energy 1 (read)           :   5.506
  HOMO energy 1 (read)           :   0.500
  LUMO energy 2 (read)           :  -3.735
  HOMO energy 2 (read)           : -11.952
 
Energy of cold fusion in kcal/mol:     NaN
 
 ==============================================
 |         Starting Energy Screening          |
 ==============================================
 
  Fast Mode selected (recommended)
  If ATM term should be included, use -atm option.
 
 Method for final opts.    : gfn1
 # of genetic optimizations: 20
 # of parents              : 100
 # of final geo. opts.     : 15
 Rare gas grid step size   :    2.50
 ang step size /deg        :   45.00
 # angular grid points     : 512
 
 Performing stack search
 Performing angular search
 
 initialization done
  Total gfn1 energy molecule 1:  -678.2371515989
  Total gfn1 energy molecule 2:  -332.9670695220
 
 -----------------------------
  Grid based energy screening 
 -----------------------------
 
  # probe RG points   :115200
  Best rare gas probe energy/kcal   :   -4.53
  +0.1 charged probe energy/kcal:  -25.25
  -0.1 charged probe energy/kcal:   -5.37
 
   Starting stack search
   Grid points: 56000
   lowest found /kcal :     NaN
 
   Starting angular search
   Grid points:33792
 
   Interaction energy of lowest structures so far in kcal/mol:
       -177.10
       -167.69
       -150.72
       -121.80
        -87.02
        -80.68
        -69.20
        -66.59
        -64.35
        -62.58
 
 ------------------------------
 genetic optimization algorithm
 ------------------------------
   cycle  Eint/kcal/mol  average Eint
    1        -210.6          NaN
    2        -101.0          NaN
    3           NaN          NaN
    4           NaN          NaN
    5           NaN          NaN
    6           NaN          NaN
    7           NaN          NaN
    8           NaN          NaN
    9           NaN          NaN
    10           NaN          NaN
    11           NaN          NaN
    12           NaN          NaN
    13           NaN          NaN
    14           NaN          NaN
    15           NaN          NaN
    16           NaN          NaN
    17           NaN          NaN
    18           NaN          NaN
    19           NaN          NaN
    20           NaN          NaN
 
Optimizing 15 best structures with gfn1
           1
           2
           3
           4
           5
           6
           7
           8
           9
          10
          11
          12
          13
          14
          15
 
   ---------------------------
      Interaction energies
   ---------------------------
 Attention: monomers are not optimized
 Interaction energies are not physical
 
  Lowest Interaction Energy:    -198.30 kcal/mol
  #   E_int (kcal/mol)
  1      -198.30
  2      -197.78
  3      -194.85
  4      -194.80
  5      -193.43
  6      -191.42
  7      -188.28
  8      -187.52
  9      -185.28
  10      -183.50
  11      -143.53
  12       -65.09
  13        -7.35
  14        -4.40
  15        83.01

------------------------------------------------------------------------
 * finished run on 2023/03/08 at 18:20:11.249     
------------------------------------------------------------------------
 total:
 * wall-time:     2 d,  1 h,  3 min, 46.002 sec
 *  cpu-time:     1 d, 18 h, 12 min, 49.400 sec
 * ratio c/w:     0.860 speedup
 LMO Computation:
 * wall-time:     0 d,  0 h,  0 min, 38.370 sec
 *  cpu-time:     0 d,  0 h,  0 min, 28.360 sec
 * ratio c/w:     0.739 speedup
 SP:
 * wall-time:     0 d,  0 h,  0 min,  0.263 sec
 *  cpu-time:     0 d,  0 h,  0 min,  1.099 sec
 * ratio c/w:     4.178 speedup
 Searching:
 * wall-time:     0 d,  0 h,  4 min, 24.476 sec
 *  cpu-time:     0 d,  0 h, 19 min, 43.051 sec
 * ratio c/w:     4.473 speedup
 ANC optimizer:
 * wall-time:     2 d,  0 h, 58 min, 36.656 sec
 *  cpu-time:     1 d, 17 h, 52 min,  7.012 sec
 * ratio c/w:     0.855 speedup

normal termination of xtb
`

**Comment by cplett** [comment]
Hi Marco,
The NaNs should indeed not occur. I will try to reproduce and fix this, but it would be best if you could provide me with the structures or, if this is not possible, something similar.
Regarding the unexpected outcome of the second POM addition: I would at least expect that it is oriented parallel to the first one on the opposite side of the CD. This might actually be a bug indicated by the NaNs. Moreover, it could be a result of an inaccurate xTB-IFF energy or the structure screening that is done with the rigid monomers. For additional output, the `--verbose` flag can be used. 
Maybe it would help to define the complete second POM for the directed docking to increase the attractive interaction also to the first POM.
